frontend_v2/src/app/components/nav/header-static/header-static.component.ts

```diff
@@ -107,6 +118,27 @@ export class HeaderStaticComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.updateElements();
     this.checkInnerWidth();
+
+    this.router.events
+    .pipe(filter(event => event instanceof NavigationEnd))
+    .subscribe((event) => {
+      if(event) {
+          if(this.router.url.split('/')[length] == "all") {
+            this.tabHighlight = "allChallenges";
+            this.globalService.changeTabActiveStatus("allChallenges");
+          }
+          else if(this.router.url.split('/')[1] == "profile") {
+            this.tabHighlight = "profile";
+            this.globalService.changeTabActiveStatus("profile");
+          }
+      }
+    });
+    this.isChallengeComponent = isNaN(this.router.url.split('/')[length]);
```

Comment:
Try `isNaN(parseInt(this.router.url.split('/')[length]));`  in order to get rid of the compilation error

Reply:
Yes, actually I didn't notice that error.